docs/src/test-parallel-js.md

```diff
@@ -57,7 +69,99 @@ You can control the maximum number of parallel worker processes via [command lin
 
 Each worker process is assigned a unique sequential index that is accessible through the [`workerInfo`](./test-advanced.md#workerinfo) object. Since each worker is a process, there also is a process-wide environment variable `process.env.TEST_WORKER_INDEX` that has the same value.
 
-## Shards
+## Group tests with serial mode
+
+If you have a few tests that depend on each other, for example one test writes to a file and another reads from that file, use the serial mode for these tests.
+
+Serial mode is enabled for a group of tests in a single file with the [`method: Test.describe.serial`] method. All tests in this serial group will be run sequentially one after another. If one of the tests fails, all subsequent tests are skipped. All tests in the group are retried together.
+
+:::note
+It is usually better to make your tests isolated, so they can be run and retried independently.
+:::
+
+You can use serial mode to implement "fail fast" strategy for a group of tests, so that all tests in the group are skipped after the first failure. For a global "fail fast" strategy, [limit the number of failures](#limit-failures).
+
+```js js-flavor=js
+const { test } = require('@playwright/test');
+
+test.describe.serial('group', () => {
+  test('runs first', async ({ page }) => {
+  });
+  test('runs second', async ({ page }) => {
+  });
+});
+```
+
+```js js-flavor=ts
+import { test } from '@playwright/test';
+
+test.describe.serial('group', () => {
+  test('runs first', async ({ page }) => {
+  });
+  test('runs second', async ({ page }) => {
+  });
+});
+```
+
+### Reusing single page between tests
+
+Playwright Test creates an isolated [Page] object for each test. However, if you'd like to reuse a single [Page] object between multiple tests, you can create your own in the [`method: Test.beforeAll`] and close it in [`method: Test.afterAll`].
+
+```js js-flavor=js
+// example.spec.js
+// @ts-check
+
+const { test } = require('@playwright/test');
+
+/** @type {import('@playwright/test').Page} */
+let page;
+
+test.beforeAll(async ({ browser }) => {
```

Comment:
having this outside of `test.describe.serial` blows my mind, can it be put inside describe block in this example?